kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -338,7 +338,14 @@ def foundry_prove(
             _LOGGER.info(f'Expanding macros in initial state for test: {test}')
             init_term = kcfg.get_unique_init().cterm.kast
             init_term = KDefinition__expand_macros(foundry.kevm.definition, init_term)
-            kcfg.replace_node(kcfg.get_unique_init().id, CTerm(init_term))
+            init_cterm = CTerm(init_term)
+
+            _LOGGER.info(f'Computing definedness constraint for initial state for test: {test}')
+            with KCFGExplore(foundry.kevm, port=find_free_port(), bug_report=br) as kcfg_explore:
+                init_cterm = kcfg_explore.cterm_assume_defined(init_cterm)
+
+            kcfg.replace_node(kcfg.get_unique_init().id, init_cterm)
+
```

Comment:
It would be good to just have a single `with KCFGExplore ...` if we need to do both definedness conditions and simplification, because starting up the RPC servers takes time.

Reply:
Good point, fixed.